File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        i = cnt = temp = 0

        for j in range(len(nums)):
            temp += nums[j]

            while temp * (j - i + 1) >= k:
                temp -= nums[i]
                i += 1

            cnt += j - i + 1
        return cnt

File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.java
class Solution {
    public long countSubarrays(int[] nums, long k) {
        int i = 0, n = nums.length; long ans = 0, temp = 0;

        for (int j = 0; j < n; j++) {
            temp += nums[j];

            while (temp * (j - i + 1) >= k) {
                temp -= nums[i++];
            }
            ans += j - i + 1;

        }
        return ans;
    }
}